ProfilerGUI/Source/Shared/ProcessOutput.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+
+namespace ProfilerGUI.Source.Shared
+{
+    /// <summary>
+    /// Data class that encapsulates the result of running a process.
+    /// </summary>
+    public class ProcessResult
+    {
+        /// <summary>
+        /// Return code the process emitted.
+        /// </summary>
+        public int ReturnCode { get; }
+
+        /// <summary>
+        /// Text printed to stdout.
+        /// </summary>
+        public string StdOut { get; }
+
+        /// <summary>
+        /// Text printed to stderr.
+        /// </summary>
+        public string StdErr { get; }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        public ProcessResult(int returnCode, string stdout = null, string stderr = null)
+        {
+            ReturnCode = returnCode;
+            StdOut = stdout ?? string.Empty;
+            StdErr = stderr ?? string.Empty;
```

Comment:
use empty string already as default param?

Reply:
I had to use a literal `""` then. I wasn't sure if that was the canonical way to do it at the time I wrote the code